src/Parser/DisciplinePriest/Modules/Features/AtonementAttribution.js

```diff
@@ -0,0 +1,30 @@
+import Module from 'Parser/Core/Module';
+import isAtonement from '../Core/isAtonement';
+
+class AtonementAttribution extends Module {
+
+  reorderEvents(events) {
+    // Push down atonement healing on self to the next atonement healing
+    let movesToMake = [];
+
+    for(let i = 0; i < events.length; i++){
+      if(events[i].type === "heal" && isAtonement(events[i]) && events[i].sourceID == events[i].targetID) {
+        // Find next atonement healing and move it right before it
+        for(let j = i + 1; j < events.length; j++) {
+          if(events[j].type == "heal" && isAtonement(events[j])) {
+            movesToMake.push({"From": i, "To": j - 1})
+            break;
```

Comment:
Ok so if I understand this correctly it moves (=prepares the instruction to move) Atonement self heal to the next Atonement? I think this is fragile and may break, such as if there's only 1 atonement active; on self: damage 1 -> damage 2 -> self heal 1 -> self heal 2 would reorder self heal 1 after 2, incorrectly. How about just breaking the loop if the first atonement heal found is a self heal?

Reply:
Tip: write a test for this situation before fixing this bug to ensure it's actually an issue.

Reply:
"damage 1 -> damage 2 -> self heal 1 -> self heal 2 would reorder self heal 1 after 2, incorrectly"
It woudn't, it would hit the next atonement, and place it right before it. (It moves from i (the self ato) to j - 1 (j behing the next atonement event, so j - 1 is right before it)

I know TDD, I actually did do the test beforehand :P

Reply:
In that case could you prove this with a test? 😇